zirgen/Dialect/ZHLT/IR/Dialect.cpp

```diff
@@ -118,4 +118,14 @@ std::string getTapsConstName() {
   return "tapList";
 }
 
+bool isEntryPoint(ComponentOp component) {
+  StringRef name = component.getName();
+  return (name.starts_with("test$") || name.ends_with("$accum") || name == "Top");
+}
+
+bool isBufferComponent(ComponentOp component) {
```

Comment:
I was kind of trying to consolidate this kind of special-casing so it wouldn't be spread around quite so much, and tried to move a bunch of it into `BufferAnalysis.cpp`.  Would that be a good place for this or is is this one more ZHLT-specific?

Reply:
I added this as a way to distinguish components that generate a buffer from those that don't before layout generation. It's used by the new `GenerateCheckLayoutPass` to know which components get a `CheckLayoutFuncOp`, and is used as an assertion in `GenerateLayoutPass` to make sure we generate a layout for the appropriately named buffer for all components.

Unfortunately, I don't see an "obvious" way to consolidate this with `BufferAnalysis`. I think the best bet there is to factor these things up to the dialect as utilities like `isEntryPoint`, `isBufferComponent`, and maybe someday a `bufferNameForComponent` (moving this out of `BufferAnalysis`). This feels a bit out of scope for this change, but those are my thoughts anyway.